src/main/java/duke/command/TaskList.java

```diff
@@ -0,0 +1,242 @@
+    private String[] arr;
+
+    public TaskList() {
+        arr = new String[100];
+    }
+
+    /**
+     * Initialize a new TaskList object
+     * @param arr
+     */
+    public TaskList(String[] arr) {
+        this.arr = arr;
+    }
+
+    /**
+     * Returns the tasklist in array
+     * @return the tasklist in array
+     * @throws IOException if array does not exist
+     */
+    public String[] readTaskList() throws IOException {
+        try {
+            try {
+                return this.arr;
+            } catch (IndexOutOfBoundsException e) {
+                throw new MissingContentException();
+            }
+        } catch (MissingContentException e) {
+            System.out.println(e.getMessage());
+        }
+        return this.arr;
+    }
+
+    /**
+     * Checks if task list is empty or not
+     * @return true if task list is empty, false otherwise
+     */
+    public boolean isEmpty() {
+        return this.getValidLen() == 0;
+    }
+
+    /**
+     * Lists tasks inside tasklist.
+     *
+     * @return task one-by-one.
+     */
+    public String list() {
+        String res = "";
+        for (int j = 0; j < arr.length; j++) {
+            if (arr[j] == null) {
+                break;
+            }
+            res += String.format("%d.%s", j + 1, arr[j]);
+            res += "\n";
+        }
+        return("Here are the tasks in your list:" + "\n" + res);
+    }
+
+    /**
+     * Returns the real length of the array.
+     * Maximum index in which array at such index is not null.
+     *
+     * @return the first null index.
+     */
+    public int getValidLen() {
+        int len = 0;
+        while (arr[len] != null) {
+            len++;
+        }
+        return len;
+    }
+
+    /**
+     * Returns new task list.
+     * Marks task at given index as done.
+     *
+     * @param num index at which task need to be marked as done.
+     * @return new task list with task marked.
+     */
+    public String mark(int num) throws IOException {
+        try {
+            try {
+                if (arr[num] != null) {
+                    String original = arr[num];
+                    arr[num] = new Task(String.valueOf(original.charAt(1)),
+                            original.substring(7), true).toString();
+                    return("OK, I've marked this task as done:" + "\n" + arr[num]);
+                }
+            } catch (IndexOutOfBoundsException e) {
+                throw new InvalidIndexException();
+            }
+        } catch (InvalidIndexException e) {
+            return(e.getMessage());
+        }
+        return "";
+    }
+
+    /**
+     * Returns new task list.
+     * Unmarks task at given index as done.
+     *
+     * @param num1 index at which task need to be marked as undone.
+     * @return new task list with task unmarked.
+     * @throw InvalidIndexException if array at specific index is null
+     */
+    public String unmark(int num1) throws IOException {
```

Comment:
Any particular reason you chose to use "num1" as the argument name? Seems a bit inconsistent with the other methods that simply use "num". Also, having a number in a variable name might be a bit unreadable.